pymc/logprob/abstract.py

```diff
@@ -152,4 +173,36 @@ def __init__(self, scalar_op, *args, **kwargs):
         super().__init__(scalar_op, *args, **kwargs)
 
 
-MeasurableVariable.register(MeasurableElemwise)
+def get_measure_type_info(
+    base_var,
+):
+    from pymc.logprob.utils import DiracDelta
+
+    if not isinstance(base_var, MeasurableVariable):
+        base_op = base_var.owner.op
+        index = base_var.owner.outputs.index(base_var)
+    else:
+        base_op = base_var
+    if not isinstance(base_op, MeasurableVariable):
+        raise TypeError("base_op must be a RandomVariable or MeasurableVariable")
+
+    if isinstance(base_op, DiracDelta):
+        ndim_supp = 0
+        supp_axes = ()
+        measure_type = MeasureType.Discrete
+        return ndim_supp, supp_axes, measure_type
+
+    if isinstance(base_op, RandomVariable):
+        ndim_supp = base_op.ndim_supp
+        supp_axes = tuple(range(-ndim_supp, 0))
+        measure_type = (
+            MeasureType.Continuous if base_op.dtype.startswith("float") else MeasureType.Discrete
+        )
+        return base_op.ndim_supp, supp_axes, measure_type
+    else:
+        # We'll need this for operators like scan and IfElse
+        if isinstance(base_op.ndim_supp, tuple):
+            if len(base_var.owner.outputs) != len(base_op.ndim_supp):
+                raise NotImplementedError("length of outputs and meta-properties is different")
```

Comment:
This is too restrictive for Scan, which can have recurrent outputs that are not measurable variables. Let's just remove it for now?